File: solutions/0168_excel_sheet_column_title/0168_excel_sheet_column_title.py
#!/usr/bin/env python3


class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        symbols = "abcdefghijklmnopqrstuvwxyz"
        title = ""

        while columnNumber:
            title = symbols[(columnNumber % 26) - 1].upper() + title
            columnNumber -= 1
            columnNumber //= 26

        return title

File: solutions/0168_excel_sheet_column_title/0168_excel_sheet_column_title.rs
impl Solution {
    pub fn convert_to_title(column_number: i32) -> String {
        let mut symbols: Vec<_> = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".chars().collect();
        let mut title: String = "".to_string();
        let mut column: i32 = column_number;
        let mut index: usize;

        while column != 0 {
            if column % 26 == 0 {
                index = 25;
            } else {
                index = ((column % 26) - 1) as usize;
            }
            title.push(symbols[index]);
            print!("{}", index);

            column -= 1;
            column /= 26;
        }
        println!();

        return title.chars().rev().collect();
    }
}
